px4_ros2_cpp/include/px4_ros2/navigation/experimental/global_navigation_interface.hpp

```diff
@@ -0,0 +1,77 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#pragma once
+
+#include <optional>
+#include <rclcpp/rclcpp.hpp>
+#include <Eigen/Eigen>
+
+#include <px4_msgs/msg/aux_global_position.hpp>
+#include <px4_ros2/navigation/experimental/navigation_interface_base.hpp>
+#include <px4_ros2/navigation/experimental/navigation_interface_common.hpp>
+
+using namespace Eigen;
+using namespace px4_msgs::msg;
+
+namespace px4_ros2
+{
+
+struct GlobalPositionEstimate
+{
+  uint64_t timestamp_sample {};
+
+  // Lat lon
+  std::optional<Vector2f> lat_lon {std::nullopt};
+
+  // Altitude (AGL frame)
+  std::optional<float> altitude_agl {std::nullopt};
+
+  // Lat lon alt standard deviation
+  std::optional<float> positional_uncertainty {std::nullopt};
+};
+
+class GlobalNavigationInterface : public NavigationInterfaceBase<GlobalPositionEstimate>
+{
+public:
+  explicit GlobalNavigationInterface(Context & context);
+  ~GlobalNavigationInterface() override = default;
+
+  /**
+   * @brief Publish global position estimate to FMU.
+   */
+  NavigationInterfaceReturnCode update(const GlobalPositionEstimate & global_position_estimate)
+  const override;
+
+  const std::string AUX_GLOBAL_POSITION_TOPIC = "/fmu/in/aux_global_position";
+
+private:
+  /**
+   * @brief Check that at least one estimate value is defined.
+   */
+  bool _checkEstimateEmpty(const GlobalPositionEstimate & estimate) const override;
+
+  /**
+   * @brief Check that if an estimate value is defined, its variance is also defined and strictly greater than zero.
+   */
+  bool _checkVarianceValid(const GlobalPositionEstimate & estimate) const override;
+
+  /**
+   * @brief Check that if an estimate value is defined, its associated frame is not *FRAME_UNKNOWN.
+   */
+  bool _checkFrameValid(const GlobalPositionEstimate & estimate) const override;
+
+  /**
+   * @brief Check that if an estimate value is defined, none of its fields are NAN.
+   */
+  bool _checkValuesNotNAN(const GlobalPositionEstimate & estimate) const override;
+
+  rclcpp::Node & _node;
```

Comment:
Shouldn't this stay `private`? Given that we are not expecting children to the global navigation interface

Reply:
My main thought here is that it should be in the interface base, since it is common to both global and local.
Then, if it's in the base, it would need to be protected so that the global & local children can access it.

Reply:
Ah duh, I didn't see the word base, will do